librz/core/csyscall.c

```diff
@@ -0,0 +1,96 @@
+// SPDX-FileCopyrightText: 2009-2021 pancake <pancake@nopcode.org>
+// SPDX-FileCopyrightText: 2009-2021 maijin <maijin21@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+
+static const char *syscallNumber(int n) {
+	return sdb_fmt(n > 1000 ? "0x%x" : "%d", n);
+}
+
+/**
+ * \brief Returns the syscall representation as a string
+ *
+ * Given the syscall number and address it resolves the syscall
+ * for the selected `asm.arch` and `asm.os` values and print
+ * its arguments.
+ *
+ * \param core RzCore instance
+ * \param n number of the syscall
+ * \param addr address of the syscall
+ */
+RZ_API RZ_OWN char *rz_core_syscall_as_string(RzCore *core, st64 n, ut64 addr) {
```

Comment:
this function is such a mess :( i understand you just copy-pasted it, so don't worry

Reply:
And it seems not working, `rz_core_syscall_as_string(core, 4, 0x0)` returns `read(0)`, not `stat()`.

Reply:
maybe let's open an issue so we track that this does not work.

Reply:
Will create an issue once this merged.

Reply:
https://github.com/rizinorg/rizin/issues/2583